File: app/src/main/java/com/example/moviemvp/movie_detail/MovieDetailsPresenter.java
package com.example.moviemvp.movie_detail;

import com.example.moviemvp.model.Movie;

public class MovieDetailsPresenter implements MovieDetailsContract.Presenter, MovieDetailsContract.Model.OnFinishedListener {
    private MovieDetailsContract.View movieDetailView;
    private MovieDetailsContract.Model movieDetailsModel;

    MovieDetailsPresenter(MovieDetailsContract.View movieDetailView) {
        this.movieDetailView = movieDetailView;
        this.movieDetailsModel = new MovieDetailsModel();
    }

    @Override
    public void onDestroy() {
        movieDetailView = null;
    }

    @Override
    public void requestMovieData(int movieId) {

        movieDetailsModel.getMovieDetails(this, movieId);
    }

    @Override
    public void onFinished(Movie movie) {
        movieDetailView.setDataToViews(movie);
    }

    @Override
    public void onFailure(Throwable t) {

        movieDetailView.onResponseFailure(t);
    }
}

File: app/src/main/java/com/example/moviemvp/movie_detail/MovieDetailsPresenter.kt
package com.example.moviemvp.movie_detail

import com.example.moviemvp.model.Movie

class MovieDetailsPresenter internal constructor(private var movieDetailView: MovieDetailsContract.View?) : MovieDetailsContract.Presenter, MovieDetailsContract.Model.OnFinishedListener {
    private val movieDetailsModel: MovieDetailsContract.Model

    init {
        this.movieDetailsModel = MovieDetailsModel()
    }

    override fun onDestroy() {
        movieDetailView = null
    }

    override fun requestMovieData(movieId: Int) {
        movieDetailsModel.getMovieDetails(this, movieId)
    }

    override fun onFinished(movie: Movie) {
        movieDetailView!!.setDataToViews(movie)
    }

    override fun onFailure(t: Throwable) {
        movieDetailView!!.onResponseFailure(t)
    }
}
